pkg/apps/strategy/support/pod.go

```diff
@@ -0,0 +1,149 @@
+package support
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/openshift/origin/pkg/api/apihelpers"
+	appsapi "github.com/openshift/origin/pkg/apps/apis/apps"
+	"github.com/openshift/origin/pkg/util"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+	kapi "k8s.io/kubernetes/pkg/apis/core"
+)
+
+func createHookPod(hook *appsapi.LifecycleHook, rc *kapi.ReplicationController, strategy *appsapi.DeploymentStrategy, suffix string,
+	startTime time.Time) (*corev1.Pod, error) {
+
+	exec := hook.ExecNewPod
+
+	baseContainer := &corev1.Container{}
+
+	for _, container := range rc.Spec.Template.Spec.Containers {
+		if container.Name == exec.ContainerName {
+			if err := legacyscheme.Scheme.Convert(&container, baseContainer, nil); err != nil {
```

Comment:
@tnozicka i will stop this crazy by converting the RC or switching this to use external client